proposals/automated-go-ipfs-releases.md

```diff
@@ -0,0 +1,121 @@
+
+Additionally, it would be great if our release process was such that we felt comfortable having an option for go-ipfs to automatically update on start instead of making users update. This would also be a step towards removing user barriers with package managers like `snap` which are not CLI tool friendly.
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+<!--(bullet list)-->
+- Doing go-ipfs releases manually must impede release cycle time in a way that impacts project velocity, or
+- The way in which users currently update go-ipfs is sufficiently an impediment that either it causes users to have a bad experience with the stack (e.g. by running into bugs which have long been fixed), or it significantly harms development velocity by needing to take into account old non-upgraded nodes
+
+#### User workflow example
+_How would a developer or user use this new capability?_
+<!--(short paragraph)-->
+
+A go-ipfs maintainer could run a script or GitHub action that would automatically perform a release from a given branch instead of manually performing the release.
+
+A go-ipfs user would be able to set an option in their config file such that on boot they would be able to automatically update the software they were running. Additionally, package managers with restrictive environments could just distribute shims around go-ipfs and the shim would be able to update and maintain itself.
+
+#### Impact
+_How would this directly contribute to web3 dev stack product-market fit?_
+
+<!--
+Explain how this addresses known challenges or opportunities.
+What awesome potential impact/outcomes/results will we see if we nail this project?
+-->
+
+Users having up to date versions of go-ipfs will give them a better experience resulting in fewer complaints about long solved issues, lower ecosystem burden and an easier time updating protocols.
+
+#### Leverage
+_How much would nailing this project improve our knowledge and ability to execute future projects?_
+
+<!--
+Explain the opportunity or leverage point for our subsequent velocity/impact (e.g. by speeding up development, enabling more contributors, etc)
+-->
+
+Future projects would make it to users faster and there would be less time from developers distracted by release process.
+
+#### Confidence
+_How sure are we that this impact would be realized? Label from [this scale](https://medium.com/@nimay/inside-product-introduction-to-feature-priority-using-ice-impact-confidence-ease-and-gist-5180434e5b15)_.
+
+<!--Explain why this rating-->
+
+
+## Project definition
+#### Brief plan of attack
+
+<!--Briefly describe the milestones/steps/work needed for this project-->
+
+#### What does done look like?
+_What specific deliverables should completed to consider this project done?_
+
+It takes less than 1hr to do a go-ipfs release from when the release is ready to be tagged, and users can automatically update to the latest version of go-ipfs.
```

Comment:
We may need to use.. IPFS :trollface:  for automatic updates to work in China:

> fwupd, the Linux firmware update service used by Lenovo, Google, Acer, Dell and more, now supports IPFS. Specifically, it was added to improve performance of firmware update transfers out of China. https://blogs.gnome.org/hughsie/2021/02/16/fwupd-1-5-6/

Reply:
I assume this is 1 person 1 hr?  I assume Bifront/infra time is separate?

Also, what goes into getting a "release ready to be tagged"?  Is that manual effort, if so should this proposal have that in scope so we drive that down?